mungegithub/github/github.go

```diff
@@ -630,14 +658,26 @@ func (config *Config) needsBranchProtection(branch *github.Branch, contexts []st
 
 // SetBranchProtection protects a branch and sets the required contexts
 func (config *Config) SetBranchProtection(name string, contexts []string) error {
-	branch, resp, err := config.client.Repositories.GetBranch(config.Org, config.Project, name)
-	config.analytics.GetBranch.Call(config, resp)
+	prot, resp, err := config.client.Repositories.GetBranchProtection(
+		context.Background(),
+		config.Org,
+		config.Project,
+		name,
+	)
+	config.analytics.GetBranchProtection.Call(config, resp)
 	if err != nil {
-		glog.Errorf("Got error getting branch %s: %v", name, err)
-		return err
+		b, errRead := ioutil.ReadAll(resp.Body)
```

Comment:
It doesn't actually cost a token since we would always have it cached. I'd always suggest code read-ability before a couple tokens. But, even I figured out what you were doing, so it's required to make it more obvious. I'd probably rather you changed it, but whatever.